include/deal.II/base/utilities.h

```diff
@@ -1425,9 +1443,16 @@ namespace Utilities
          const std::vector<char>::const_iterator &cend,
          const bool                               allow_compression)
   {
-    // see if the object is small and copyable via memcpy. if so, use
-    // this fast path. otherwise, we have to go through the BOOST
-    // serialization machinery
+    // see if the object is empty. if so, shortcut the whole
+    // operation and return an empty object.  this avoids
+    // dereferencing iterators if the std::vector<char> is empty,
+    // which would not be allowed.
+    if (cbegin == cend)
+      return T();
```

Comment:
same

Reply:
This should be ok -- we create the to-be-returned object below via
```
  T object;
```
so we already require that the object be default-constructible.

Reply:
But I think that some clang-tidy or other check will ask that you write this instead:
```suggestion
      return {};
```